projects/rocprofiler-sdk/source/include/rocprofiler-sdk/agent.h

```diff
@@ -117,6 +117,31 @@ ROCPROFILER_CXX_CODE(
         sizeof(rocprofiler_agent_runtime_visiblity_t) == sizeof(uint32_t),
         "Increasing the size of the rocprofiler_agent_runtime_visiblity_t is not permitted");)
 
+/**
+ * @brief Firmware version information for a GPU agent.
+ *
+ * This struct contains the firmware version numbers for various GPU microcontrollers and engines.
+ * All fields are 32-bit unsigned integers representing the firmware version as reported by the
+ * corresponding sysfs files in /sys/class/drm/renderD{}/device/fw_version/
+ */
+typedef struct rocprofiler_agent_fw_info_t
+{
```

Comment:
We  are moving to versioning for all public structs. Include a size element as the first element here similar to agent_v0

```suggestion
{typedef struct rocprofiler_agent_v0_t
{
    uint64_t size;  ///< set to sizeof(rocprofiler_agent_t) by rocprofiler. This can be used for
                    ///< versioning and compatibility handling
```